File: 算法/Python/第5章 动态规划/5.1 动态规划/stairs_dp.py
def climb_stairs(n):
    if n <= 0:
        return 0
    elif n == 1:
        return 1
    elif n == 2:
        return 2
    num1 = 1
    num2 = 2
    for _ in range(3, n+1):
        sum = num1 + num2
        num1 = num2
        num2 = sum
    return sum

def main():
    print(climb_stairs(10))

if __name__ == '__main__':
    main()

File: code/Java/Chapter12/12-1/StairsDp.java
public class StairsDp {
    public static void main(String[] args) {
        System.out.println(climbStairs(10));
    }

    public static int climbStairs(int n) {
        if(n <= 0) {
            return 0;
        } else if(n == 1) {
            return 1;
        } else if(n == 2) {
            return 2;
        }
        int num1 = 1;
        int num2 = 2;
        int sum = 0;
        for(int i = 3; i <= n; i++) {
            sum = num1 + num2;
            num1 = num2;
            num2 = sum;
        }
        return sum;
    }
}